Marketplace/MarketplaceToolkit.ps1

```diff
@@ -0,0 +1,2109 @@
+                </ScrollViewer>
+            </Grid>
+        </DockPanel>
+    </Grid>
+</Window>
+'@
+#endregion
+
+#region Get XAML and create variables
+Add-Type -AssemblyName PresentationFramework
+
+[XML]$XML = $XAML -replace "x:N",'N' 
+
+$Reader = (New-Object System.XML.XMLNodeReader $XML)
+$Form = [Windows.Markup.XAMLReader]::Load($Reader)
+
+# Create variables > Load XAML Objects In PowerShell
+$XML.SelectNodes("//*[@Name]") | % {Set-Variable -Name "X_$($_.Name)" -Value $Form.FindName($_.Name)}
+
+$timer = new-object 'System.Windows.Threading.DispatcherTimer'
+$timer.Interval = [TimeSpan]"0:0:1.00"
+
+#endregion
+
+#region Functions
+function F_Regex {
+$X_Input_Tbx_Name.Tag = @{'regex'='[ a-zA-Z0-9]{1,80}';'errormessage'='Maximum 80 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Publisher.Tag = @{'regex'='[ a-zA-Z0-9]{1,20}';'errormessage'='Maximum 30 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Summary.Tag = @{'regex'='[ a-zA-Z0-9]{1,100}';'errormessage'='Maximum 100 characters. Can only contain A-Z, a-z, 0-9 and spaces.'}
+$X_Input_Tbx_Description.Tag = @{'regex'='.{1,5000}';'errormessage'='Maximum 5000 characters.'}
+$X_Input_Tbx_Category.Tag = @{'regex'='[- a-zA-Z0-9]{1,64}';'errormessage'='Maximum 64 characters. Can only contain A-Z, a-z, 0-9, - and spaces.'}
+$X_Input_Tbx_Icon40.Tag = @{'errormessage'='The image size must be 40x40.'}
+$X_Input_Tbx_Icon90.Tag = @{'errormessage'='The image size must be 90x90.'}
+$X_Input_Tbx_Icon115.Tag = @{'errormessage'='The image size must be 115x115.'}
+$X_Input_Tbx_Icon255.Tag = @{'errormessage'='The image size must be 255x115.'}
+$X_Input_Tbx_Screenshot.Tag = @{'errormessage'='The image size must be 533x324.'}
+$X_Params_Tbx_AddStep.Tag = @{'regex'='[ a-zA-Z]{1,20}';'errormessage'='Maximum 20 characters. Can only contain A-Z, a-z and spaces.'}
+$X_Publish_Pwb_Password1.Tag = @{'errormessage'='Passwords do not match.'}
+$X_Publish_Pwb_Password1.Tag = @{'errormessage'='Passwords do not match.'}
+}
+
+function F_Clear {
+$X_Input_Tbx_ParamFile.Text = $null
+$X_Input_Tbx_Name.Text = $null
+$X_Input_Tbx_Publisher.Text = $null
+$X_Input_Tbx_SUmmary.Text = $null
+$X_Input_Tbx_description.Text = $null
+$X_Input_Tbx_category.Text = $null
+$X_Input_Tbx_Icon40.Text = $null
+$X_Input_Tbx_Icon90.Text = $null
+$X_Input_Tbx_Icon115.Text = $null
+$X_Input_Tbx_Icon255.Text = $null
+$X_Input_Tbx_Screenshot.Text = $null
+$X_Preview_Img_Icon40.Source = $null
+$X_Preview_Img_Screenshot.Source = $null
+$X_Params_Tbx_Template.Text = $null
+$X_Params_Tbx_Exe.Text = $null
+$X_Params_Tree_View.Tag = $null
+$X_Params_Tree_View.Items.Clear()
+$X_Params_Bdr_Steps.Visibility = 'Collapsed'
+$X_Publish_Tbx_Package.Text = $null
+$X_Publish_Tbx_Username.Text = $null
+$X_Publish_Pwb_Password1.Password = $null
+$X_Publish_Pwb_Password2.Password = $null
+$X_Publish_Tbx_Endpoint.Text = $null
+$X_Publish_Lsv_Log.Items.Clear()
+$timer.Stop()
+}
+
+function F_Validation {
```

Comment:
Based on the parameters specified in an eventhandler a validation is triggered is in this function. Not sure how the parameter set would help here. Can you explain how?